benchmark/benchmark.cpp

```diff
@@ -0,0 +1,354 @@
+        if (ret != UMF_RESULT_SUCCESS) {
+            return;
+        }
+
+        // those function should never fail, so error handling is minimal.
+        ret = umfDisjointPoolParamsSetSlabMinSize(disjoint_memory_pool_params,
+                                                  4096);
+        if (ret != UMF_RESULT_SUCCESS) {
+            goto err;
+        }
+
+        ret = umfDisjointPoolParamsSetCapacity(disjoint_memory_pool_params, 4);
+        if (ret != UMF_RESULT_SUCCESS) {
+            goto err;
+        }
+
+        ret = umfDisjointPoolParamsSetMinBucketSize(disjoint_memory_pool_params,
+                                                    4096);
+        if (ret != UMF_RESULT_SUCCESS) {
+            goto err;
+        }
+
+        ret = umfDisjointPoolParamsSetMaxPoolableSize(
+            disjoint_memory_pool_params, 4096 * 16);
+
+        if (ret != UMF_RESULT_SUCCESS) {
+            goto err;
+        }
+        return;
+    err:
+
+        umfDisjointPoolParamsDestroy(disjoint_memory_pool_params);
+        disjoint_memory_pool_params = NULL;
+    }
+
+    ~disjoint_pool() {
+        if (disjoint_memory_pool_params != NULL) {
+            umfDisjointPoolParamsDestroy(disjoint_memory_pool_params);
+        }
+    }
+
+    umf_memory_pool_ops_t *
+    getOps([[maybe_unused]] ::benchmark::State &state) override {
+        return umfDisjointPoolOps();
+    }
+    void *getParams([[maybe_unused]] ::benchmark::State &state) override {
+
+        if (disjoint_memory_pool_params == NULL) {
+            state.SkipWithError("Failed to create disjoint pool params");
+        }
+
+        return disjoint_memory_pool_params;
+    }
+    static std::string name() {
+        return "disjoint_pool<" + Provider::name() + ">";
+    }
+};
+#endif
+
+#ifdef UMF_BUILD_LIBUMF_POOL_JEMALLOC
+template <typename Provider>
+struct jemalloc_pool : public pool_interface<Provider> {
+    umf_memory_pool_ops_t *
+    getOps([[maybe_unused]] ::benchmark::State &state) override {
+        return umfJemallocPoolOps();
+    }
+    void *getParams([[maybe_unused]] ::benchmark::State &state) override {
+        return NULL;
+    }
+    static std::string name() {
+        return "jemalloc_pool<" + Provider::name() + ">";
+    }
+};
+#endif
+
+template <typename Provider>
+struct scalable_pool : public pool_interface<Provider> {
+    virtual umf_memory_pool_ops_t *
+    getOps([[maybe_unused]] ::benchmark::State &state) override {
+        return umfScalablePoolOps();
+    }
+    virtual void *
+    getParams([[maybe_unused]] ::benchmark::State &state) override {
+        return NULL;
+    }
+    static std::string name() {
+        return "scalable_pool<" + Provider::name() + ">";
+    }
+};
+
+// Benchmarks scenarios:
+
+UMF_BENCHMARK_TEMPLATE_DEFINE(alloc_benchmark, glibc_fix, fixed_alloc_size,
+                              glibc_malloc);
+
+UMF_BENCHMARK_REGISTER_F(alloc_benchmark, glibc_fix)
```

Comment:
Add comment on what each parameter does. Also, what does it mean to pass Args twice? and Threads?

Reply:
Add comment on what each parameter does. - Done

 Also, what does it mean to pass Args twice? and Threads - this is documented in google benchmark docs (or header file). It does run benchmark in all possible configurations. In this case we have 4 configs created by caesarean product between arguments and threads set.